File: Backend/src/main/java/com/project/autoserve/service/MechanicProfileService.java
package com.project.autoserve.service;

import com.project.autoserve.dto.profile.MechanicProfileResponseDTO;
import com.project.autoserve.dto.profile.UpdateProfileRequestDTO;

public interface MechanicProfileService {

    MechanicProfileResponseDTO getProfile();

    MechanicProfileResponseDTO updateProfile(
            UpdateProfileRequestDTO request);

}

File: Frontend/src/services/mechanicProfileService.js
import axiosInstance from "../api/axiosInstance";

export const mechanicProfileService = {

    getProfile: async () => {

        const response =
            await axiosInstance.get("/mechanic/profile");

        return response.data;
    },

    updateProfile: async (profileData) => {

        const response =
            await axiosInstance.put(
                "/mechanic/profile",
                profileData
            );

        return response.data;
    }

};